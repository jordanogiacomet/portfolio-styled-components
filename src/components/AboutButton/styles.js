import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
