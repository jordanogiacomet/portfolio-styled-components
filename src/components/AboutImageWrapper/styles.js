import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
