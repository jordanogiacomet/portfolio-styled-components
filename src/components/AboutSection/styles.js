import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
