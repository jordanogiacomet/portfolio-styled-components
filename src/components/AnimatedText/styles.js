import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;