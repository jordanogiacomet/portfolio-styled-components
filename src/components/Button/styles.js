import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 