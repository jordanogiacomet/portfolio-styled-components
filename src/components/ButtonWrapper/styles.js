import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;