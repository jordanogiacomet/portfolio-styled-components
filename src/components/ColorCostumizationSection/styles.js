// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
