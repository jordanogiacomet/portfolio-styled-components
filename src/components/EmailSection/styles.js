import styled from 'styled-components';

export const Container = styled.section`
  display: grid;
  grid-template-columns: repeat(1, 1fr);

  @media ${({ theme }) => theme.media.lg} {
    grid-template-columns: repeat(12, 1fr);
    align-items: center;
    justify-content: space-between;
    gap: 5rem;
  }
`;

export const Col = styled.div`
  grid-column: span 1;
  @media ${({ theme }) => theme.media.sm} {
    grid-column: ${({ span }) => `span ${span}`};
  };

  &.self-center {
    place-self: center;
  };

  &.text-center {
    text-align: center;
  };

  &.form-element{
    @media(max-width: 1024px) {
      order: 2;
    };
  }

  &.text-element{
    text-align: right !important;
    margin-left: 30px;
    @media(max-width: 1024px) {
      order: 1;
      margin-left: 0px;
      margin-bottom: 40px;
      text-align: left !important;
    };
  }

  @media ${({ theme }) => theme.media.sm} {
    text-align: left;
  };

  &.mt-4 {
    margin-top: 5rem;

    @media ${({ theme }) => theme.media.lg} {
      margin-top: 0;
    }
  };
`;

export const FormDiv = styled.div`
  margin-bottom: 2rem;
`;