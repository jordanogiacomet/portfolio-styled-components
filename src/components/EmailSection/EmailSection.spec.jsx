import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import { EmailSection } from '.';
import '@testing-library/jest-dom/extend-expect';
import { ColorProvider } from '../../contexts/ColorCustomizationContext';
import { ColorModeProvider } from '../../contexts/ColorModeContext';
import { LanguageProvider } from '../../contexts/LanguageContext'; // Certifique-se de que o LanguageProvider est√° importado

describe('<EmailSection />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render heading', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const heading = screen.getByRole('heading', { name: /connect|conectar/i });
    expect(heading).toBeInTheDocument();
  });

  it('should render email input', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const emailInput = screen.getByPlaceholderText('user@example.com');
    expect(emailInput).toBeInTheDocument();
  });

  it('should render subject input', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const subjectInput = screen.getByPlaceholderText('Just saying hi');
    expect(subjectInput).toBeInTheDocument();
  });

  it('should render message textarea', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const messageTextArea = screen.getByPlaceholderText('Lets talk about...');
    expect(messageTextArea).toBeInTheDocument();
  });

  it('should render send button', () => {
    renderTheme(
      <ColorModeProvider>
        <ColorProvider>
          <LanguageProvider>
            <EmailSection />
          </LanguageProvider>
        </ColorProvider>
      </ColorModeProvider>
    );
    const sendButton = screen.getByRole('button', { name: /send message|enviar mensagem/i });
    expect(sendButton).toBeInTheDocument();
  });
});
