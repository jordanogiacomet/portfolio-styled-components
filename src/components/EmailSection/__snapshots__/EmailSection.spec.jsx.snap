// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailSection /> should match snapshot 1`] = `
.c1 {
  grid-column: span 1;
}

.c1.self-center {
  place-self: center;
}

.c1.text-center {
  text-align: center;
}

.c1.text-element {
  text-align: right!important;
  margin-left: 30px;
}

.c1.mt-4 {
  margin-top: 5rem;
}

.c8 {
  grid-column: span 1;
}

.c8.self-center {
  place-self: center;
}

.c8.text-center {
  text-align: center;
}

.c8.text-element {
  text-align: right!important;
  margin-left: 30px;
}

.c8.mt-4 {
  margin-top: 5rem;
}

.c11 {
  font-size: 1.5rem;
  color: #FFFFFF;
  margin-bottom: 20px;
  margin-top: 20px;
}

.c13 {
  height: 3.5rem;
  width: 3.5rem;
  border: 0.125rem solid #adb7be;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  transition: border-color 0.3s;
  visibility: hidden;
  opacity: 0;
}

.c13.mail-icons {
  opacity: 1!important;
  visibility: visible!important;
}

.c13.mail-icons.light {
  color: white;
  background: white;
}

.c13.mail-icons.dark {
  color: white;
  background: white;
}

.c13:hover {
  border-color: white;
}

.c13:hover .icon {
  color: white;
}

.c13 .icon {
  height: 2.5rem;
  width: 2.5rem;
  color: #adb7be;
  transition: color 0.3s;
}

.c12 {
  display: flex;
  flex-direction: row;
  gap: 20px;
  justify-content: right;
}

.c10 {
  background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c9 {
  color: #ADB7BE;
  font-size: 3.2rem;
  text-transform: none;
  z-index: 2;
}

.c2 {
  display: flex;
  flex-direction: column;
}

.c4 {
  color: #FFFFFF;
  display: block;
  margin-bottom: 1rem;
  font-size: 1.5rem;
  font-weight: 600;
}

.c4.light {
  color: #FFFFFF;
}

.c5 {
  background: #18191E;
  border: 1px solid #33353F;
  color: #9CA2A9;
  font-size: 1.5rem;
  border-radius: 0.5rem;
  width: 100%;
  padding: 1.5rem;
  display: block;
}

.c5:focus {
  outline: none;
  border-color: #1E90FF;
}

.c5::placeholder {
  color: #9CA2A9;
  font-size: 1.5rem;
}

.c6 {
  background-color: #18191E;
  border: 1px solid #33353F;
  color: #9CA2A9;
  font-size: 1.5rem;
  border-radius: 0.5rem;
  display: block;
  width: 100%;
  padding: 1.5rem;
  resize: vertical;
  height: 200px;
}

.c6::placeholder {
  color: #9CA2A9;
  font-size: 1.5rem;
}

.c6:focus {
  outline: none;
  border-color: #1E90FF;
}

.c7 {
  background: linear-gradient(to bottom right, #1E90FF, #4169E1, #00008B);
  border: transparent;
  color: white;
  font-weight: 600;
  padding: 1rem 1.25rem;
  border-radius: 10rem;
  width: 100%;
  transition: scale 0.2s ease-in-out;
  cursor: pointer;
}

.c7:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c7:hover {
  transition: scale ease-in-out 0.2s;
  scale: calc(1.02);
}

.c7:focus {
  outline: none;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  margin-bottom: 100px;
}

.c3 {
  margin-bottom: 2rem;
}

@media (min-width: 640px) {
  .c1 {
    grid-column: span 4;
  }
}

@media (max-width: 1024px) {
  .c1.form-element {
    order: 2;
  }
}

@media (max-width: 1024px) {
  .c1.text-element {
    order: 1;
    margin-left: 0px;
    margin-bottom: 40px;
    text-align: left!important;
  }
}

@media (min-width: 640px) {
  .c1 {
    text-align: left;
  }
}

@media (min-width: 1024px) {
  .c1.mt-4 {
    margin-top: 0;
  }
}

@media (min-width: 640px) {
  .c8 {
    grid-column: span 8;
  }
}

@media (max-width: 1024px) {
  .c8.form-element {
    order: 2;
  }
}

@media (max-width: 1024px) {
  .c8.text-element {
    order: 1;
    margin-left: 0px;
    margin-bottom: 40px;
    text-align: left!important;
  }
}

@media (min-width: 640px) {
  .c8 {
    text-align: left;
  }
}

@media (min-width: 1024px) {
  .c8.mt-4 {
    margin-top: 0;
  }
}

@media (min-width: 640px) {
  .c11 {
    font-size: 1.8rem;
  }
}

@media (min-width: 1024px) {
  .c11 {
    font-size: 2rem;
  }
}

@media (max-width: 1024px) {
  .c12 {
    justify-content: left;
  }
}

@media (max-width: 640px) {
  .c0 {
    margin-bottom: 30px;
  }
}

@media (min-width: 1024px) {
  .c0 {
    grid-template-columns: repeat(12, 1fr);
    align-items: center;
    justify-content: space-between;
    gap: 5rem;
  }
}

<div>
  <div
    id="contact"
  >
    <section
      class="c0"
      type="email-section"
    >
      <div
        class="c1 form-element"
        span="4"
      >
        <form
          class="c2"
        >
          <div
            class="c3"
            type="form-container"
          >
            <label
              class="c4 dark"
              color="#FFFFFF"
              for="email"
            >
              Your email
            </label>
            <input
              class="c5"
              id="email"
              name="email"
              placeholder="user@example.com"
              required=""
              type="email"
            />
          </div>
          <div
            class="c3"
            type="form-container"
          >
            <label
              class="c4 dark"
              color="#FFFFFF"
              for="subject"
            >
              Subject
            </label>
            <input
              class="c5"
              id="subject"
              name="subject"
              placeholder="Just saying hi"
              required=""
              type="text"
            />
          </div>
          <div
            class="c3"
            type="form-container"
          >
            <label
              class="c4 dark"
              color="#FFFFFF"
              for="message"
            >
              Your message
            </label>
            <textarea
              class="c6"
              id="message"
              name="message"
              placeholder="Lets talk about..."
            />
          </div>
          <div
            class="c3"
            type="form-container"
          >
            <button
              class="c7"
              from="#1E90FF"
              to="#00008B"
              type="submit"
              via="#4169E1"
            >
              Send message
            </button>
          </div>
        </form>
      </div>
      <div
        class="c8 text-element"
        span="8"
      >
        <h5
          class="c9"
        >
          <span
            class="c10"
            from="#1E90FF"
            to="#00008B"
            via="#4169E1"
          >
            Let's connect
          </span>
        </h5>
        <p
          class="c11"
          color="#FFFFFF"
          type="default"
        >
          I am currently looking for new opportunities. Please feel free to contact me and I will respond as quickly as possible.
        </p>
        <div
          class="c12"
        >
          <a
            class="c13 mail-icons dark"
            href="https://github.com/jordanogiacomet"
            target="_blank"
          >
            <img
              alt="Github icon"
              loading="lazy"
              src="github-icon.svg"
            />
          </a>
          <a
            class="c13 mail-icons dark"
            href="https://www.linkedin.com/in/jordano-giacomet-tomazoni-aba346184/"
            target="_blank"
          >
            <img
              alt="Linkedin icon"
              loading="lazy"
              src="linkedin-icon.svg"
            />
          </a>
        </div>
      </div>
    </section>
  </div>
</div>
`;
